src/deluge/dsp/reverb/mutable/reverb.hpp

```diff
@@ -0,0 +1,152 @@
+// Copyright 2023 Katherine Whitlock
+// Copyright 2014 Emilie Gillet
+// Reverb.
+
+#pragma once
+#include "dsp/reverb/base.hpp"
+#include "fx_engine.hpp"
+#include <array>
+#include <limits>
+#include <numeric>
+#include <span>
+
+namespace deluge::dsp::reverb {
+
+class Mutable : public Base {
+	constexpr static size_t kBufferSize = 32768;
+
+public:
+	Mutable() : reverb_time_(0.35f + 0.63f * 0.5f) {}
+
+	~Mutable() override = default;
+
+	void Process(std::span<int32_t> in, std::span<StereoSample> output) override {
+		// This is the Griesinger topology described in the Dattorro paper
+		// (4 AP diffusers on the input, then a loop of 2x 2AP+1Delay).
+		// Modulation is applied in the loop of the first diffuser AP for additional
+		// smearing; and to the two long delays for a slow shimmer/chorus effect.
+		typename FxEngine::AllPass ap1(150);
+		typename FxEngine::AllPass ap2(214);
+		typename FxEngine::AllPass ap3(319);
+		typename FxEngine::AllPass ap4(527);
+
+		typename FxEngine::AllPass dap1a(2182);
+		typename FxEngine::AllPass dap1b(2690);
+		typename FxEngine::AllPass del1(4501);
+
+		typename FxEngine::AllPass dap2a(2525);
+		typename FxEngine::AllPass dap2b(2197);
+		typename FxEngine::AllPass del2(6312);
+
+		typename FxEngine::Context c;
+		FxEngine::ConstructTopology(engine_, //<
+		                            {
+		                                &ap1, &ap2, &ap3, &ap4, //<
+		                                &dap1a, &dap1b, &del1,  //<
+		                                &dap2a, &dap2b, &del2,  //<
+		                            });
+
+		const float kap = diffusion_;
+		const float klp = lp_;
+		const float krt = reverb_time_;
+		const float gain = input_gain_;
+
+		float lp_1 = lp_decay_1_;
+		float lp_2 = lp_decay_2_;
+
+		for (size_t frame = 0; frame < in.size(); frame++) {
+			StereoSample& s = output[frame];
+			float wet = 0;
+			float apout = 0.0f;
+			engine_.Advance();
+
+			// Smear AP1 inside the loop.
+			// c.Interpolate(ap1, 10.0f, LFO_1, 80.0f, 1.0f);
+			// c.Write(ap1, 100, 0.0f);
```

Comment:
Is this intentionally commented out?

Reply:
yes, see [Rings code](https://github.com/pichenettes/eurorack/blob/08460a69a7e1f7a81c5a2abcc7189c9a6b7208d4/rings/dsp/fx/reverb.h#L93)